From 01d1a0954ab58711a0de55dc42921ad36eeb9c97 Mon Sep 17 00:00:00 2001
From: David Edmundson <kde@davidedmundson.co.uk>
Date: Fri, 28 Mar 2025 09:34:43 +0000
Subject: [PATCH] notifications: Sync window position on creation

QWindow::moveEvent only are fired when there is a platform window.
This means we need to sync the position on platform window creation.
---
 applets/notifications/notificationwindow.cpp | 12 ++++++++++++
 applets/notifications/notificationwindow.h   |  1 +
 2 files changed, 13 insertions(+)

diff --git a/applets/notifications/notificationwindow.cpp b/applets/notifications/notificationwindow.cpp
index 025523361b..29650785ca 100644
--- a/applets/notifications/notificationwindow.cpp
+++ b/applets/notifications/notificationwindow.cpp
@@ -65,6 +65,18 @@ void NotificationWindow::setIsCritical(bool critical)
     Q_EMIT isCriticalChanged();
 }
 
+bool NotificationWindow::event(QEvent *e)
+{
+    // moveEvents only occur when we have a platform window, so we need to explicitly sync on creation
+    if (e->type() == QEvent::PlatformSurface) {
+        auto ps = static_cast<QPlatformSurfaceEvent *>(e);
+        if (ps->surfaceEventType() == QPlatformSurfaceEvent::SurfaceCreated) {
+            PlasmaShellWaylandIntegration::get(this)->setPosition(QPoint(x(), y()));
+        }
+    }
+    return PlasmaWindow::event(e);
+}
+
 void NotificationWindow::moveEvent(QMoveEvent *me)
 {
     PlasmaShellWaylandIntegration::get(this)->setPosition(me->pos());
diff --git a/applets/notifications/notificationwindow.h b/applets/notifications/notificationwindow.h
index 38685bdca8..cdae8451dc 100644
--- a/applets/notifications/notificationwindow.h
+++ b/applets/notifications/notificationwindow.h
@@ -23,6 +23,7 @@ public:
     void setIsCritical(bool critical);
 
 protected:
+    bool event(QEvent *e) override;
     void moveEvent(QMoveEvent *) override;
 Q_SIGNALS:
     void takeFocusChanged();
-- 
GitLab

